psi4/src/psi4/ccenergy/ccenergy.cc

```diff
@@ -654,4 +860,444 @@ void CCEnergyWavefunction::one_step(void) {
     return;
 }
 
+void CCEnergyWavefunction::get_t1_rhf(SharedMatrix &_t1)
+{
+  dpdfile2 tIA;
+  global_dpd_->file2_init(&tIA, PSIF_CC_OEI, 0, 0, 1, "tIA");
+  global_dpd_->file2_mat_init(&tIA);
+  global_dpd_->file2_mat_rd(&tIA);
+
+  for(int h = 0; h < moinfo_.nirreps; ++h)
+  {
+    int upper_bound_ij = moinfo_.occpi[h] - moinfo_.openpi[h];
+    int upper_bound_IA = moinfo_.virtpi[h] - moinfo_.openpi[h];
+
+    for(int i = 0; i < moinfo_.occpi[h]; ++i)
+    {
+      int I = i + moinfo_.frdocc[h];
+      for(int a = 0; a < upper_bound_IA; ++a)
+      {
+        int A = a + moinfo_.frdocc[h] + moinfo_.occpi[h];
+        _t1->set(h, I, A, tIA.matrix[h][i][a]);
+      }
+    }
+  }
+
+  global_dpd_->file2_close(&tIA);
+}
```

Comment:
There is a Matrix function that reads in `dpdfile2`'s. Can you use that?

Edit: Same comment for the next several functions. It might be wise to edit the Matrix function if its not completely up to snuff.